### PR TITLE
Enable Karaf console and password by default

### PR DESCRIPTION
**Pull request by mstormi**
Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: functions/openhab.bash [hunk 1]
```diff
@@ -143,12 +143,12 @@ openhab_shell_interfaces() {
...
       if ! sshPass2="$(whiptail --title "Authentication setup" --passwordbox "\\nPlease confirm the password:" 9 80 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
-      if [[ $sshPass1 == "$sshPass2" ]] && [[ ${#sshPass1} -ge 8 ]] && [[ ${#sshPass2} -ge 8 ]]; then
+      if [[ $sshPass1 == "$sshPass2" ]] && [[ ${#sshPass1} -ge 7 ]] && [[ ${#sshPass2} -ge 7 ]]; then
         sshPass="$sshPass1"
       else
-        whiptail --title "Authentication setup" --msgbox "Password mismatched, blank, or less than 8 characters... Please try again!" 7 80
+        whiptail --title "Authentication setup" --msgbox "Password mismatched, or less than 7 characters... Please try again!" 7 80
       fi
     done
   fi
```
Comment (ecdye) [inline functions/openhab.bash:154]: This will not work, it still requires interactive setup with the current code.

File: functions/openhab.bash [hunk 2]
```diff
@@ -158,7 +158,6 @@ openhab_shell_interfaces() {
 
   if ! cond_redirect sed -i -e 's|^sshHost = 127.0.0.1.*$|sshHost = 0.0.0.0|g' /var/lib/openhab/etc/org.apache.karaf.shell.cfg; then echo "FAILED (sshHost)"; return 1; fi
   if cond_redirect sed -i -e 's|openhab = .*,|openhab = '"${sshPass}"',|g' /var/lib/openhab/etc/users.properties; then echo "OK"; else echo "FAILED (sshPass)"; return 1; fi
-  cond_redirect systemctl restart openhab.service
```
Comment (ecdye) [inline functions/openhab.bash:161]: Are you sure it is OK to remove this?

**Review by ecdye — CHANGES_REQUESTED** [review]
Is this really a good idea to enable this by default? It would allow malicious actors take advantage of careless users who do not think to change the default password. Because this will make the Karaf console available on external interfaces this problem will be much worse than in the current implementation where a malicious actor needs direct access to the box to do any harm.

File: functions/openhab.bash (hunk 2 above)
Comment (mstormi) [reply functions/openhab.bash:161]: yes you can change those files with OH up

File: functions/openhab.bash (hunk 1 above)
Comment (mstormi) [reply functions/openhab.bash:154]: if you want to define the password yes this requires interaction (just an instant idea: we could maybe reuse userpw)
but the more important part is the editing of .cfg and that should be done unattended during install

Comment (mstormi) [reply functions/openhab.bash:154]: What about this: no password set after unattended install but if run interactively it'll deploy the default password when user does not enter anything. To manually edit sshHost is a lot more hassle to users and for remote admins it's awkward without a console.

File: openhabian-setup.sh [hunk 3]
```diff
@@ -102,6 +102,7 @@ if [[ -n "$UNATTENDED" ]]; then
   vimrc_copy
   install_tailscale "install" && setup_tailscale
   misc_system_settings
+  openhab_shell_interfaces
```
Comment (ecdye) [inline openhabian-setup.sh:105]: Shouldn't this go after openHAB is installed?

**Review by ecdye — CHANGES_REQUESTED** [review]

File: openhabian-setup.sh (hunk 3 above)
Comment (mstormi) [reply openhabian-setup.sh]: indeed, here you are

**Review by ecdye — APPROVED** [review]
LGTM, DNT

**Pull request merged**